core/include/usListenerFunctors.h

```diff
@@ -34,6 +34,25 @@ namespace us {
   class BundleEvent;
 
   typedef std::function<void(const ServiceEvent&)> ServiceListener;
+
+  /**
+   * \ingroup MicroServices
+   *
+   * A \c BundleEvent listener. When a \c BundleEvent is fired, it is
```

Comment:
The SynchronousBundleListener is necessary to implement Declarative Services.
@karthikreddy09 can you log an issue so we can track this work? Thanks.